FirebaseTestProject/Views/PostRow.swift

```diff
@@ -17,14 +20,35 @@ struct PostRow: View {
             Text(post.text)
                 .font(.body)
                 .padding()
-            Text(post.author)
-                .padding()
+            HStack {
+                Text(post.author)
+                if post.author == authViewModel.getUser() {
+                    Spacer()
+                    Button {
+                        Task {
+                            do {
+                                try await PostService.delete(post)
+                            }
+                            catch {
+                                print(error)
+                            }
+                            await postData.loadPosts()
```

Comment:
A single row shouldn't be able to load all of the posts.  Try to remove the `postData` dependency and replace it with a simple closure `onDelete` to execute when the trash icon is tapped.